src/traces/image/plot.js

```diff
@@ -144,8 +138,19 @@ module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
         // Pixelated image rendering
         // http://phrogz.net/tmp/canvas_image_zoom.html
         // https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering
-        image3
-            .attr('style', 'image-rendering: optimizeSpeed; image-rendering: -moz-crisp-edges; image-rendering: -o-crisp-edges; image-rendering: -webkit-optimize-contrast; image-rendering: optimize-contrast; image-rendering: crisp-edges; image-rendering: pixelated;');
+        var style = 'image-rendering: optimizeSpeed; image-rendering: -moz-crisp-edges; image-rendering: -o-crisp-edges; image-rendering: -webkit-optimize-contrast; image-rendering: optimize-contrast; image-rendering: crisp-edges; image-rendering: pixelated;';
+        if(fastImage) {
+            // Flip the SVG image as needed (around the proper center location)
+            var axisScale = [
+                (xa.range[0] < xa.range[1]) ? 1 : -1,
+                (ya.range[0] < ya.range[1]) ? -1 : 1
+            ];
+            style += 'transform:' +
+                'translate(50%,50%)' +
+                'scale(' + axisScale[0] + ',' + axisScale[1] + ')' +
+                'translate(-50%,-50%)' + ';';
+        }
```

Comment:
Actually we cannot use percent as the position is not correct e.g. if I remove the margin block from `image_astronaut_source`.
Please require `strTranslate` again and replace the block with this exact setup.
```js
        if(fastImage) {
            var flipX = xa.range[1] < xa.range[0];
            var flipY = ya.range[1] > ya.range[0];
            if(flipX || flipY) {
                var tx = left + imageWidth / 2;
                var ty = top + imageHeight / 2;
                style += 'transform:' +
                    strTranslate(tx + 'px', ty + 'px') +
                    'scale(' + (flipX ? -1 : 1) + ',' + (flipY ? -1 : 1) + ')' +
                    strTranslate(-tx + 'px', -ty + 'px') + ';';
            }
        }
```
Thanks.
Then we need to figure out a way to adjust the hover for the cases the (x|y) scale is negative.

Reply:
update: The hover would also be correct in the browser if you apply this change.
But to lock some issues I noticed, I suggest we duplicate image_astronaut_source as a new mock and replace the `layout` with 
```js
"layout": {
        "width": 500,
        "height": 400,
        "yaxis": {
            "scaleanchor": "x",
            "range": ["64", "512"]
        },
        "xaxis": {
            "autorange": "reversed"
        },
        "margin": {
            "t": 50,
            "b": 30
        }
    }
```

Reply:
Noting that in JavaScript "64" > "512", we need to convert raw axis ranges to linear form.
So please update the block to be:
```js
        if(fastImage) {
            var xRange = Lib.simpleMap(xa.range, xa.r2l);
            var yRange = Lib.simpleMap(ya.range, ya.r2l);

            var flipX = xRange[1] < xRange[0];
            var flipY = yRange[1] > yRange[0];
            if(flipX || flipY) {
                var tx = left + imageWidth / 2;
                var ty = top + imageHeight / 2;
                style += 'transform:' +
                    strTranslate(tx + 'px', ty + 'px') +
                    'scale(' + (flipX ? -1 : 1) + ',' + (flipY ? -1 : 1) + ')' +
                    strTranslate(-tx + 'px', -ty + 'px') + ';';
            }
        }
```

Reply:
Instead of duplicating `image_astronaut_source` which is huge, let's adjust your original mock removed in https://github.com/plotly/plotly.js/pull/5307/commits/a19b05b8253b371a17dbc5632f5b5db6bcbe34a5 to have similar ranges e.g.
```js
"yaxis3": {"range": ["8", "32"]},
```